docs/dependency-injection/dependency-injection.md

```diff
@@ -0,0 +1,220 @@
+# What is Dependency Injection (DI)?
+
+Dependency injection (DI) is a design pattern that provides loosely-coupled, maintainable, and testable code. It improves modularity, maintainability, and testability, and is easily customizable and extensible.
+
+In the context of Oryx, dependency injection allows users to customize logic deep inside the framework, which is particularly useful for projects with complex or rapidly-evolving requirements. Without DI, users would need to override large portions of the logic or create a lot of boilerplate code. By leveraging DI, users can override logic while still being able to upgrade to newer versions of Oryx going forward.
+
+One key advantage of using Oryx's DI implementation is that it is vanilla JavaScript and can be used in other frameworks as well. Although there are standard DI packages available, using Oryx's implementation ensures seamless integration and compatibility with its features.
```

Comment:
> `Although there are standard DI packages available`

What do you mean by `standard` here? Maybe let's rephrase to `popular`?

Reply:
👍